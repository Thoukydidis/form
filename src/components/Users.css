.usersList {
  max-height: 41%;
  overflow: scroll;
}
