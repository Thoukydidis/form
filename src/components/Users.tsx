import { Button } from "@mui/material";
import React, { useState } from "react";
import SearchIcon from "@mui/icons-material/Search";
import { UsersResponse } from "../utils/interfaces/index";
import { fetchUsers } from "../utils/api";
import "./Users.css";

export interface Props {
  setSelectedUser: React.Dispatch<React.SetStateAction<number | undefined>>;
  selectedUser: number | undefined;
}
const Users = (props: Props) => {
  const { setSelectedUser, selectedUser } = props;
  const [users, setUsers] = useState<UsersResponse[]>();

  const getUsers = async () => {
    try {
      const response: any = await fetchUsers();
      setUsers(response);
    } catch (e: any) {
      console.log(JSON.stringify(e));
    }
  };
  const handleClick = (id: number) => {
    setSelectedUser(id);
  };

  const displayUsers = (users?: UsersResponse[]) => {
    const userLabel = users?.map((user: UsersResponse, index: number) => {
      return (
        <div>
          <button
            data-testid="getUsersButton"
            onClick={() => handleClick(user.id)}
            style={
              selectedUser === user.id
                ? { display: "flex", width: "100%", backgroundColor: "gray" }
                : { display: "flex", width: "100%" }
            }
          >
            <p style={{ padding: "7px" }}>{user.id}</p>
            <p style={{ padding: "7px" }}>{user.name}</p>
          </button>
        </div>
      );
    });
    return userLabel && <div className="usersList">{userLabel}</div>;
  };
  console.log("selectedUser:", selectedUser);
  return (
    <div style={{ width: "15%", marginRight: "15px" }}>
      <h1>Users</h1>
      <Button onClick={getUsers} data-testid="convertButton">
        Get Users
        <SearchIcon />
      </Button>
      {displayUsers(users)}
    </div>
  );
};

export default Users;
