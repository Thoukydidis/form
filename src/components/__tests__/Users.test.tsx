import React from "react";
import Users, { Props } from "../Users";
import { cleanup, render, fireEvent, screen } from "@testing-library/react";

describe("test the Users component", () => {
  afterEach(cleanup);
  const props: Props = {
    setSelectedUser: () => null,
    selectedUser: 3070,
  };
  it("should render the Users component", () => {
    const view = render(<Users {...props} />);
    expect(view).toMatchSnapshot();
  });
  it("should render the histoyBox styles", () => {
    const firstInitialState = [
      {
        id: 3684,
        name: "Ganak Varma",
        email: "varma_ganak@stanton.io",
        gender: "female",
        status: "inactive",
      },
    ];
    jest
      .spyOn(React, "useState")
      .mockImplementationOnce(() => [firstInitialState, () => null]);

    const view = render(<Users {...props} />);
    const converButton = screen.getByTestId("getUsersButton");
    expect(converButton).toBeTruthy();
    fireEvent.click(converButton);
    expect(view).toMatchSnapshot();
  });
});
