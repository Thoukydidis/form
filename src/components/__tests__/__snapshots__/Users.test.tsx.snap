// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Users component should render the Users component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 15%; margin-right: 15px;"
      >
        <h1>
          Users
        </h1>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="convertButton"
          tabindex="0"
          type="button"
        >
          Get Users
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 15%; margin-right: 15px;"
    >
      <h1>
        Users
      </h1>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        data-testid="convertButton"
        tabindex="0"
        type="button"
      >
        Get Users
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
